core/vm/evm.go

```diff
@@ -31,6 +32,38 @@ import (
 // deployed contract addresses (relevant after the account abstraction).
 var emptyCodeHash = crypto.Keccak256Hash(nil)
 
+// Type of an executing call frame
+type CallFrameType int
+
+const (
+	CallType CallFrameType = iota
+	CallCodeType
+	DelegateCallType
+	StaticCallType
+	// Init code execution
+	CreateType
+	Create2Type
+)
```

Comment:
These seem not needed. We already have these operations defined as opcodes, how about just passing along the op that caused this scope to enter?